util/printchplenv

```diff
@@ -93,6 +93,7 @@ CHPL_ENVS = [
     ChapelEnv('CHPL_REGEXP', RUNTIME | DEFAULT),
     ChapelEnv('CHPL_LLVM', COMPILER, 'llvm'),
     ChapelEnv('CHPL_AUX_FILESYS', RUNTIME | DEFAULT, 'fs'),
+    ChapelEnv('CHPL_LIBMODE', RUNTIME | DEFAULT, 'libmode'),
```

Comment:
I don't think we want this to be printed out by default (i.e. I don't think we want this to be visible in the output of a simple `printchplenv`.

Reply:
Because we don't expect it to be changed often?

Reply:
We've tried to minimize how much is printed out in printchplenv by default. e.g. we don't show `CHPL_HOST_PLATFORM`, `CHPL_HOST_MEM`, `CHPL_HOST_COMPILER`, `CHPL_LLVM`, `CHPL_GMP` and maybe a few others. @ben-albrecht and/or @mppf probably know more and may have an opinion on whether `CHPL_LIBMODE` should be printed by default

Reply:
`CHPL_GMP` is printed out, as well as `CHPL_AUX_FILESYS`.  I don't have a strong opinion, though, so am happy to go with the group preference

Reply:
I'd prefer to keep it hidden for now, especially since we're trying to keep it for interop only at this point (because of perf implications and stuff)

Reply:
It's all a bit arbitrary IMO, but I am in favor of keeping fewer variables in the default output when possible. It's relatively painless to include `CHPL_LIBMODE` as a default variable in the future if we determine it's frequently used enough.

Reply:
In the email discussion, I think Greg proposed something along the lines of only including it if it got set to the "pic" value, similar to the GASNet information (I think that impacts both the build directory and the printchplenv output).  Does that seem like a good idea?

Reply:
I thought the GASNet information (e.g. SEGMENT) was hidden unless CHPL_COMM=gasnet. That seems very different from this case.

Reply:
The implementation is [here](https://github.com/chapel-lang/chapel/blob/de12350bfff956c53fd05ed9e8adb47fa457f994/util/printchplenv#L231).  I meant that the act of hiding the information under certain circumstances was similar, not that we would base it off the setting of a different environment variable

Reply:
Okay, I will follow the recommendation to not include it in the default output.  Thanks!